request/graphql/parser/filter.go

```diff
@@ -190,6 +190,55 @@ func parseVal(val ast.Value, recurseFn parseFn) (any, error) {
 	return nil, ErrFailedToParseConditionValue
 }
 
+// ParseFilterFieldsForDescription parses the fields that are defined in the SchemaDescription
+// from the filter conditions“
+func ParseFilterFieldsForDescription(
+	conditions map[string]any,
+	schema client.SchemaDescription,
+) []client.FieldDescription {
+	return parseFilterFieldsForDescriptionMap(conditions, schema)
+}
+
+func parseFilterFieldsForDescriptionMap(
+	conditions map[string]any,
+	schema client.SchemaDescription,
+) []client.FieldDescription {
+	fields := make([]client.FieldDescription, 0)
+	for k, v := range conditions {
+		switch k {
+		case "_or", "_and":
+			conds := v.([]any)
+			parsedFileds := parseFilterFieldsForDescriptionSlice(conds, schema)
+			fields = append(fields, parsedFileds...)
+		case "_not":
+			conds := v.(map[string]any)
+			parsedFileds := parseFilterFieldsForDescriptionMap(conds, schema)
+			fields = append(fields, parsedFileds...)
+		default:
+			f, found := schema.GetField(k)
+			if !found || f.IsObject() {
+				continue
+			}
+			fields = append(fields, f)
+		}
+	}
+	return fields
+}
+
+func parseFilterFieldsForDescriptionSlice(conditions []any, schema client.SchemaDescription) []client.FieldDescription {
+	fields := make([]client.FieldDescription, 0)
+	for _, v := range conditions {
+		switch cond := v.(type) {
+		case map[string]any:
+			parsedFields := parseFilterFieldsForDescriptionMap(cond, schema)
+			fields = append(fields, parsedFields...)
+		default:
+			// error
```

Comment:
todo: This doesn't add anything useful since there are no errors created and this function doesn't return an error either. I would suggest removing the default case.

Reply:
Updated to actually have an error return, so the entire call stack also has an error path